src/frontend/ir_generation/codes.cc

```diff
@@ -0,0 +1,84 @@
+/**
```

Comment:
There isn't very detailed documentation. Maybe some clarifications above the function declarations might help.